src/cryptonote_protocol/cryptonote_protocol_defs.h

```diff
@@ -344,12 +345,14 @@ namespace cryptonote
       uint64_t timestamp;
       crypto::public_key pubkey;
       crypto::signature sig;
+      epee::net_utils::ipv4_network_address storage_endpoint;
```

Comment:
It would save some rearrangement in the future (and slightly reduce the serialization size) to break this into a `uint32_t public_ip; uint16_t storage_server_port;` pair.  For now it just reconstitutes into a storage server endpoint, but in the future there will be multiple ports on that same IP.

Reply:
shouldn't these be a string so it can do ipv6?

Reply:
IPv4 is going to be a requirement already (because every node needs to be able to contact every other node).  But I think an additional (optional) IPv6 address could be supported so that the IPv6 address gets preferred if both ends are IPv6-enabled.

Reply:
Changed to using two separate fields: ip and port. I think could add IPv6 as a separate field in the future.